--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -5,7 +5,7 @@
 
 config PRINTK_TIME
 	bool "Show timing information on printks"
-	depends on PRINTK
+	depends on n
 	help
 	  Selecting this option causes time stamps of the printk()
 	  messages to be added to the output of the syslog() system
@@ -239,7 +239,7 @@
 
 choice
 	prompt "Debug information"
-	depends on DEBUG_KERNEL
+	default DEBUG_INFO_NONE
 	help
 	  Selecting something other than "None" results in a kernel image
 	  that will include debugging info resulting in a larger kernel image.
@@ -712,7 +712,7 @@
 
 config DEBUG_OBJECTS
 	bool "Debug object operations"
-	depends on DEBUG_KERNEL
+	depends on n
 	help
 	  If you say Y here, additional code will be inserted into the
 	  kernel to track the life time of various objects and validate
@@ -813,7 +813,7 @@
 
 config DEBUG_VM
 	bool "Debug VM"
-	depends on DEBUG_KERNEL
+	depends on n
 	help
 	  Enable this to turn on extended checks in the virtual-memory system
 	  that may impact performance.
@@ -891,8 +891,8 @@
 	  regions to be regularly checked for invalid topology.
 
 config DEBUG_MEMORY_INIT
-	bool "Debug memory initialisation" if EXPERT
-	default !EXPERT
+	bool
+	default n
 	help
 	  Enable this for additional checks during memory initialisation.
 	  The sanity checks verify aspects of the VM such as the memory model
@@ -1350,7 +1350,7 @@
 
 config PROVE_LOCKING
 	bool "Lock debugging: prove locking correctness"
-	depends on DEBUG_KERNEL && LOCK_DEBUGGING_SUPPORT
+	depends on n
 	select LOCKDEP
 	select DEBUG_SPINLOCK
 	select DEBUG_MUTEXES if !PREEMPT_RT
@@ -1443,7 +1443,7 @@
 
 config DEBUG_SPINLOCK
 	bool "Spinlock and rw-lock debugging: basic checks"
-	depends on DEBUG_KERNEL
+	depends on n
 	select UNINLINE_SPIN_UNLOCK
 	help
 	  Say Y here and build SMP to catch missing spinlock initialization
@@ -1453,7 +1453,7 @@
 
 config DEBUG_MUTEXES
 	bool "Mutex debugging: basic checks"
-	depends on DEBUG_KERNEL && !PREEMPT_RT
+	depends on n
 	help
 	 This feature allows mutex semantics violations to be detected and
 	 reported.
@@ -1865,7 +1865,7 @@
 	bool "Filter access to /dev/mem"
 	depends on MMU && DEVMEM
 	depends on ARCH_HAS_DEVMEM_IS_ALLOWED || GENERIC_LIB_DEVMEM_IS_ALLOWED
-	default y if PPC || X86 || ARM64
+	default y
 	help
 	  If this option is disabled, you allow userspace (root) access to all
 	  of memory, including kernel and userspace memory. Accidental