--- a/mm/Kconfig.debug	2026-05-09 21:33:11.939796514 +0800
+++ b/mm/Kconfig.debug	2026-05-09 21:40:36.082647528 +0800
@@ -46,8 +46,8 @@
 	  can be overridden by debug_pagealloc=off|on.
 
 config SLUB_DEBUG
-	default y
-	bool "Enable SLUB debugging support" if EXPERT
+	default n
+	bool "Enable SLUB debugging support"
 	depends on SYSFS && !SLUB_TINY
 	select STACKDEPOT if STACKTRACE_SUPPORT
 	help
